source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/RendererUtil.java

```diff
@@ -66,7 +68,44 @@ public static Calendar getTime(String s)
         return calendar;
     }
 
+    public static class SpecialTextHandleResult
+    {
+        private CharSequence m_htmlString;
+        private boolean m_hasLinks;
+
+        public SpecialTextHandleResult(CharSequence htmlString, boolean hasLinks)
+        {
+            m_htmlString = htmlString;
+            m_hasLinks = hasLinks;
+        }
+
+        public CharSequence getHtmlString()
+        {
+            return m_htmlString;
+        }
+
+        public boolean getHasLinks()
+        {
+            return m_hasLinks;
+        }
+    }
+
     public static CharSequence handleSpecialText(String textWithFormattedDates)
+    {
+        Spanned spanned = getSpecialTextSpans(textWithFormattedDates);
+        return trimHtmlString(spanned);
+    }
+
+    public static SpecialTextHandleResult handleSpecialTextAndQueryLinks(String textWithFormattedDates)
+    {
+        Spanned spanned = getSpecialTextSpans(textWithFormattedDates);
+
+        URLSpan[] spans = spanned.getSpans(0, spanned.length() - 1, URLSpan.class);
```

Comment:
nit: would this work with phone number link?

Reply:
should we stop at phone numbers?